test/Types/Common.hs

```diff
@@ -252,11 +232,30 @@ genWhitespace = Gen.element
 genText :: Gen Text
 genText = Gen.text ( Range.linear 0 100 ) Gen.unicodeAll
 
-parseBS :: ByteString -> Either DecodeError Json
-parseBS = SD.parseWith AB.parseOnly parseWaargonaut
+genScientific :: MonadGen m => Maybe Int -> m Scientific
+genScientific lim = either fst fst . Sci.fromRationalRepetend lim
+  <$> Gen.realFrac_ (Range.linearFrac 0.0001 1000.0)
+
+parseBS :: SD.Decoder Identity a -> ByteString -> Either (DecodeError, SD.CursorHistory) a
+parseBS d = SD.pureDecodeFromByteString AB.parseOnly d
+
+parseText :: SD.Decoder Identity a -> Text -> Either (DecodeError, SD.CursorHistory) a
+parseText d = SD.pureDecodeFromText AT.parseOnly d
+
+encodeJsonText :: Json -> Text
+encodeJsonText = TextL.toStrict . E.simplePureEncodeText E.json
+
+encodeText :: E.Encoder Identity a -> a -> TextL.Text
```

Comment:
I still feel like encodeText/encodeBS and decodeText/decodeBS ought to be in Waargonaut. Make a opinionated choice about what the standard default should be but explain how they can peel a layer off and make a different choice if needed.  But then I guess that you have do do this for both the pure and non pure versions so it gets messy real quick. :(

Reply:
I had scratched it off, but I think I will do this. It is starting to seem silly that you can't just get the package and start using it. `attoparsec` is in the gargantuan dependency tree anyway.

If I have these functions in a different module then it doesn't get too messy.